cli_FSD/api_routes.py

```
@@ -15,35 +15,117 @@
+    
+    # Filter out stop words and short words
+    keywords = [word for word in words if word not in stop_words and len(word) > 2]
+    
+    # If we filtered out too much, fall back to original query
+    if not keywords:
+        return query
+    
+    # Join keywords with spaces for better matching
+    return ' '.join(keywords)
+
 def fetch_relevant_embeddings(query, max_results=5):
```

Flake8 (7.2.0)

[error] expected 2 blank lines, found 1

(E302)